### PR TITLE
Add contour and tricontour plots to plot types

### PR DESCRIPTION
**Pull request by timhoffm**

### CODE REVIEW

**Comment by jklymak** [comment]
I think there is benefit to not cluttering this list up too much.  I'm not sure separating contour and contour is worth the extra entry(s).

**Comment by story645** [comment]
I like it seperate since I think a goal of this is making explicit implicit assumptions about data structure, in this case that it's unstructured coordinates. But kinda think this should go under the imshow/array section.

**Comment by jklymak** [comment]
tricontour/f already had its now entry.  The question here is whether we need separate entries for each of contour and contourf and tricontour and tricontourf.

**Comment by story645** [comment]
Man, reading the docs I'm even more confused about when to use the regular version vs. the f one, & that's not explained when collapsed into one example either. Does it make sense to somehow highlight that distinction/is there a way to do so briefly?

File: plot_types/unstructured/tricontourf.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+y = Y[:, 0][ysamp]
+x = X[0, :][xsamp]
+z = Z[ysamp, xsamp]
+
+# plot:
+fig, ax = plt.subplots()
+ax.grid(False)
+
+ax.plot(x, y, 'o', markersize=2, color='grey')
+levels = np.linspace(np.min(Z), np.max(Z), 7)
```
Comment (story645) [inline plot_types/unstructured/tricontourf.py:31]: Should be up w/ rest of data code?

**Review by story645 — COMMENTED** [review]
Should levels be set w/ rest of data?

**Comment by jklymak** [comment]
Again, I don't think the goal here is to be comprehensive, but rather to give folks enough knowledge to know what to search for.  In this case both `ax.contour` and `ax.contourf` link to the respective methods.  The first sentence of either method is: 
`contour and contourf draw contour lines and filled contours, respectively.`

**Comment by story645** [comment]
When I click through, the first sentence I see on both methods is 

>Draw contour lines on an unstructured triangular grid.

https://matplotlib.org/devdocs/api/_as_gen/matplotlib.pyplot.tricontourf.html?highlight=tricontourf#matplotlib.pyplot.tricontourf

which I think is because of  #17742?

And for me it's not about comprehensive, it's that contour and contourf have different use cases and collapsing them like this makes that unclear.

**Comment by jklymak** [comment]
Sure, if folks think they are distinct enough, they should get separate entries.  I don't think they are, but that is just my opinion.

File: plot_types/unstructured/tricontourf.py (hunk 1 above)
Comment (timhoffm) [reply plot_types/unstructured/tricontourf.py]: Done.

**Comment by timhoffm** [comment]
I think they are distinct enough visually.

IMHO the main use case of "Plot types" is to look at the plots and decide which visual representation you want. And for that lines and filled areas are quite different. That they encode the same type of information is secondary. It's somewhat analogous to `stem()` encoding the same as `bar()` (or even `plot()` with marker only, but certainly you won't trade it easily for the others and would want to see that all of them exist.

**Comment by timhoffm** [comment]
Note: I had to rename: examples/frontpage/contour.py → examples/frontpage/contour_frontpage.py

Apparently, Sphinx-gallery doesn't like the same name in different galleries: 

> WARNING: Duplicate example file name(s) found. Having duplicate file names will break some links. List of files: ['../plot_types/arrays/contour.py']

While the above renaming breaks someone external linking the front page coutour example, I anticipate that that'd be avery rare case and I willing to take that so that we have more consistent names in the `plot_types`.

**Review by story645 — APPROVED** [review]

**Comment by story645** [comment]
Needs a rebate needs a rebase cause of #21123

**Pull request merged**